core/src/main/java/hudson/model/Label.java

```diff
@@ -427,7 +427,34 @@ public boolean contains(Node node) {
      * This is usually used as a signal that this label is invalid.
      */
     public boolean isEmpty() {
-        return getNodes().isEmpty() && getClouds().isEmpty();
+        return hasNoNodes() && hasNoClouds();
```

Comment:
For readability I would change this to

```suggestion
        return hasAnyNodes() && hasAnyClouds();
```

and adapt the rest of the code accordingly, but 🤷